fiftyone/server/db/_mongo_lightning.py

```
@@ -0,0 +1,627 @@
+
+    collection = foo.get_async_db_conn()[collection]
+
+    while isinstance(field, fof.ListField):
+        field = field.field
+
+    if not isinstance(field, (fof.ObjectIdField, fof.StringField)) and (
+        path.exclude or path.search
+    ):
+        raise ValueError(
+            f"'excluded' and 'search' are not valid for {fof.StringField}"
+        )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)